### PR TITLE
feat(vertical-pod-autoscaler): Support recommender only deployment

### PR DESCRIPTION
**Pull request by agershman**
Supports a mode of deployment where only the recommender is deployed and the updater and admission controller and not. This provides for a cleaner deployment as opposed to the current state where you can only scale these deployments down to 0. E.g. In the case that alerting has been setup, special casing isn't needed to account for a deployment that's meant to have replicas set at 0.

### CODE REVIEW

File: charts/vertical-pod-autoscaler/values.yaml [hunk 1]
```diff
@@ -19,6 +19,8 @@ serviceMonitor:
 
 logLevel: 4
 
+recommenderOnly: false
```
Comment (agershman) [inline charts/vertical-pod-autoscaler/values.yaml:22]: Defaulting to existing behavior.

File: charts/vertical-pod-autoscaler/templates/admission-controller/certificate.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/admission-controller/certificate.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/admission-controller/pdb.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/admission-controller/pdb.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/admission-controller/serviceaccount.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/admission-controller/serviceaccount.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/admission-controller/servicemonitor.yaml [hunk 5]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/admission-controller/servicemonitor.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/updater/pdb.yaml [hunk 6]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/updater/pdb.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/updater/service.yaml [hunk 7]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/updater/service.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/updater/serviceaccount.yaml [hunk 8]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/updater/serviceaccount.yaml:1]: Please add this condition to the `if` below.

File: charts/vertical-pod-autoscaler/templates/updater/servicemonitor.yaml [hunk 9]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.recommenderOnly -}}
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/templates/updater/servicemonitor.yaml:1]: Please add this condition to the `if` below.

**Review by stevehipwell — CHANGES_REQUESTED** [review]
Thanks for the PR @agershman. This makes sense but I've added some minor syntactical change requests. Could you also update the README and add a new CHANGELOG entry.

**Comment by agershman** [comment]
> Thanks for the PR @agershman. This makes sense but I've added some minor syntactical change requests. Could you also update the README and add a new CHANGELOG entry.

Sure thing. I'll get those addressed and amend. Thank you!

File: charts/vertical-pod-autoscaler/README.md [hunk 10]
```diff
@@ -69,7 +69,8 @@ The following table lists the configurable parameters of the _Vertical Pod Autos
...
 | `admissionController.certManager.enabled`                | If `true`, use _Cert Manager_ to create and manage the certificates for the webhook.                                                                                                             | `false`                                  |
 | `admissionController.certManager.issuerKind`             | The type of issuer that `admissionController.certManager.issuerName` refers to.                                                                                                                  | `Issuer`                                 |
-| `admissionController.certManager.issuerName`             | If set, the _Cert Manager_ certificate will be configued to use this issuer.                                                                                                                     | `null`                                   |
+| `admissionController.certManager.issuerName`             | If set, the _Cert Manager_ certificate will be configured to use this issuer.                                                                                                                    |
```
Comment (agershman) [inline charts/vertical-pod-autoscaler/README.md:72]: Fixed a misspelling here.

File: charts/vertical-pod-autoscaler/CHANGELOG.md [hunk 11]
```diff
@@ -20,6 +20,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [UNRELEASED]
 
+## [v1.5.0] - 2024-01-05
+
```
Comment (stevehipwell) [inline charts/vertical-pod-autoscaler/CHANGELOG.md:24]:
```suggestion
```

The version will be set when this is released.

**Review by stevehipwell — CHANGES_REQUESTED** [review]
Just a minor change required.

File: charts/vertical-pod-autoscaler/CHANGELOG.md (hunk 11 above)
Comment (stevehipwell) [reply charts/vertical-pod-autoscaler/CHANGELOG.md:24]: Could you please remove this line?

**Review by stevehipwell — CHANGES_REQUESTED** [review]
@agershman if you can remove the version line from CHANGELOG we can run the tests and hopefully get this merged.

File: charts/vertical-pod-autoscaler/CHANGELOG.md (hunk 11 above)
Comment (agershman) [reply charts/vertical-pod-autoscaler/CHANGELOG.md]: All set.

**Review by stevehipwell — APPROVED** [review]
LGTM

**Pull request merged**